pkg/pod/entrypoint_test.go

```diff
@@ -651,11 +721,23 @@ func TestStopSidecars(t *testing.T) {
 		Name:  injectedSidecar.Name,
 		Image: nopImage,
 	}
+	// This is a container that is added by the controller for accessing sidecar logs.
+	// This should not be stopped as long as results-from is set to sidecar-logs.
```

Comment:
Can we inject the sidecar only if results-from is set to sidecar-logs?

Reply:
Yes, this is already the case : https://github.com/tektoncd/pipeline/blob/963aa40829045a349c8f78d4fa3dd10ffca46282/pkg/pod/pod.go#L155

It will also ensure that the sidecar is only injected if the task is producing results. It will not blindly inject a sidecar for all taskruns.